### PR TITLE
Pass regular children to ReflexContainer.

### PR DESCRIPTION
**Pull request by rpiaggio**
`ReflexContainer` expects a regular list of react children, so we can pass that instead of an array to avoid having to provide keys for the elements.

### CODE REVIEW

**Review by cquiroz — APPROVED** [review]
LGTM

File: observationtree/src/main/scala/explore/observationtree/ConstraintSetObsList.scala [hunk 1]
```diff
@@ -304,39 +304,39 @@ object ConstraintSetObsList {
...
+                        ),
+                        Icons.ChevronRight
+                      )
 
-                        val obsSelected = props.focused.get
-                          .exists(f => csObs.map(obs => FocusedObs(obs.id)).exists(f === _))
+                      val obsSelected = props.focused.get
+                        .exists(f => csObs.map(obs => FocusedObs(obs.id)).exists(f === _))
 
+                      <.span(^.key := csId.toString)(
```
Comment (toddburnside) [inline observationtree/src/main/scala/explore/observationtree/ConstraintSetObsList.scala:339]: Do we need this `span` along with the other changes you made?

**Review by toddburnside — APPROVED** [review]
LGTM

File: observationtree/src/main/scala/explore/observationtree/ConstraintSetObsList.scala (hunk 1 above)
Comment (rpiaggio) [reply observationtree/src/main/scala/explore/observationtree/ConstraintSetObsList.scala:339]: Good catch. I was just trying things out here. I'll remove it, it's not necessary.

**Pull request merged**